src/components/config_parser.cpp

```diff
@@ -144,6 +144,11 @@ void config_parser::parse_file(const string& file, file_list path) {
 
     if (!line.is_header && line.key == "include-file") {
       parse_file(file_util::expand(line.value), path);
+    } else if (!line.is_header && line.key == "include-directory") {
+        const string expanded_path = file_util::expand(line.value);
+        for (const auto& filename : file_util::list_files(expanded_path)) {
```

Comment:
We should sort the list here. This gives us a consistent include order and will also prevent issues when one of the files does not with a section header.